tiatoolbox/tools/tissuemask.py

```diff
@@ -0,0 +1,289 @@
+        >>> thumbnail = imread("thumbnail.png")
+        >>> masker = MorphologicalMasker(power=1.25)
+        >>> masks = masker.fit_transform([thumbnail])
+
+    """
+
+    def __init__(
+        self, *, mpp=None, power=None, kernel_size=None, min_region_size=None
+    ) -> None:
+        """Initialise a morphological masker.
+
+        Args:
+            mpp (float or tuple(float)):
+                The microns per-pixel of the image to be masked. Used to
+                calculate kernel_size a 64/mpp, optional.
+            power (float or tuple(float)):
+                The objective power of the image to be masked. Used to
+                calculate kernel_size as 64/objective_power2mpp(power),
+                optional.
+            kernel_size (int or tuple(int)):
+                Size of elliptical kernel in x and y, optional.
+            min_region_size (int):
+                Minimum region size in pixels to consider as foreground.
+                Defaults to area of the kernel.
+        """
+        super().__init__()
+
+        self.min_region_size = min_region_size
+        self.threshold = None
+
+        # Check for conflicting arguments
+        if sum(arg is not None for arg in [mpp, power, kernel_size]) > 1:
+            raise ValueError("Only one of mpp, power, kernel_size can be given.")
+
+        # Default to kernel_size of (1, 1) if no arguments given
+        if all(arg is None for arg in [mpp, power, kernel_size]):
+            kernel_size = np.array([1, 1])
+
+        # Convert (objective) power approximately to MPP to unify units
+        if power is not None:
+            mpp = objective_power2mpp(power)
+
+        # Convert MPP to an integer kernel_size
+        if mpp is not None:
+            mpp = np.array(mpp)
+            if mpp.size != 2:
+                mpp = mpp.repeat(2)
+            kernel_size = np.max([32 / mpp, [1, 1]], axis=0)
+
+        # Ensure kernel_size is a length 2 numpy array
+        kernel_size = np.array(kernel_size)
+        if kernel_size.size != 2:
+            kernel_size = kernel_size.repeat(2)
+
+        # Convert to an integer double/ pair
+        self.kernel_size = tuple(np.round(kernel_size).astype(int))
+
+        # Create structuring element for morphological operations
+        self.kernel = cv2.getStructuringElement(cv2.MORPH_ELLIPSE, self.kernel_size)
+
+        # Set min region size to kernel area if None
+        if self.min_region_size is None:
+            self.min_region_size = np.sum(self.kernel)
+
+    def transform(self, images: np.ndarray):
+        """Create masks using the found threshold followed by morphological operations.
+
+
+        Args:
+            images (:class:`numpy.ndarray`): List of images with a
+                length 4 shape (N, height, width, channels).
+
+        Returns:
+            :class:`numpy.ndarray`: List of images with a length 4
+                shape (N, height, width, channels).
+        """
```

Comment:
Please add a new line at the end.

Reply:
Fixed in this commit https://github.com/TIA-Lab/tiatoolbox/pull/61/commits/5b0e0ecc1fcb9ce989d907483bef9a2b0b2c7aa5